tasks/request_from_archive/bin/run_tests.sh

```diff
@@ -58,13 +40,11 @@ fi
 
 python3 -m venv venv
 source venv/bin/activate
+trap 'deactivate' EXIT
 
 ## Install the requirements
-pip install -q --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org
-pip install -q -r requirements-dev.txt --trusted-host pypi.org --trusted-host files.pythonhosted.org
-let return_code=$?
-
-check_rc $return_code "ERROR: pip install encountered an error."
+run_and_check_returncode "pip install -q --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org"
```

Comment:
Should maybe do thisthroughout instead of just here. Or move to common?

Reply:
If we do it in common, the trap logic gets more complicated. Left where it is for now.

Reply:
Sorry, should have clarified. I was thinking we should move this to `create_and_activate_venv`